homeassistant/components/venstar/__init__.py

```diff
@@ -55,35 +60,74 @@ async def async_unload_entry(hass, config):
     return unload_ok
 
 
-class VenstarEntity(Entity):
-    """Get the latest data and update."""
+class VenstarDataUpdateCoordinator(update_coordinator.DataUpdateCoordinator):
+    """Class to manage fetching Venstar data."""
 
-    def __init__(self, config, client):
-        """Initialize the data object."""
-        self._config = config
-        self._client = client
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        venstar_connection: VenstarColorTouch,
+    ) -> None:
+        """Initialize global Venstar data updater."""
+        self.client = venstar_connection
 
-    async def async_update(self):
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=60),
+        )
+
+    async def _async_update_data(self) -> None:
         """Update the state."""
         try:
-            info_success = await self.hass.async_add_executor_job(
-                self._client.update_info
-            )
+            await self.hass.async_add_executor_job(self.client.update_info)
         except (OSError, RequestException) as ex:
-            _LOGGER.error("Exception during info update: %s", ex)
+            raise update_coordinator.UpdateFailed(
+                f"Exception during Venstar info update: {ex}"
+            ) from ex
 
         # older venstars sometimes cannot handle rapid sequential connections
         await asyncio.sleep(3)
 
         try:
-            sensor_success = await self.hass.async_add_executor_job(
-                self._client.update_sensors
-            )
+            await self.hass.async_add_executor_job(self.client.update_sensors)
         except (OSError, RequestException) as ex:
-            _LOGGER.error("Exception during sensor update: %s", ex)
+            raise update_coordinator.UpdateFailed(
+                f"Exception during Venstar sensor update: {ex}"
+            ) from ex
+        return None
+
+
+class VenstarEntity(CoordinatorEntity):
+    """Representation of a Venstar entity."""
+
+    def __init__(
+        self,
+        venstar_data_coordinator: VenstarDataUpdateCoordinator,
+        config: ConfigEntry,
+    ) -> None:
+        """Initialize the data object."""
+        super().__init__(venstar_data_coordinator)
+        self._config = config
+        self._update_attr()
+        self.coordinator = venstar_data_coordinator
```

Comment:
The coordinator is set by the parent class. We can remove this line.

Reply:
If I remove:
self.coordinator = venstar_data_coordinator

Then how will this work?

def _client(self):
        """Return the venstar client."""
        return self.coordinator.client

I couldn't find a way to access the client from the climate entity without setting it somewhere like this.